tools/sdk-usage/main.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+// Generating list of used GO SDK operation IDs.
+func main() {
+	argsNumber := 3
+	// Check if the folder path is provided as a command-line argument
+	if len(os.Args) != argsNumber {
+		fmt.Println("Usage: go run main.go <folder_path> <output_file>")
+		os.Exit(1)
+	}
+
+	folderPath := os.Args[1]
+	outputFile := os.Args[2]
+
+	if folderPath == "" {
+		fmt.Println("Please provide a folder path")
+		os.Exit(1)
+	}
+	if outputFile == "" {
+		fmt.Println("Please provide an output file")
+		os.Exit(1)
+	}
+
+	// Define the regular expression pattern
+	pattern := `s\.clientv2\.[\w\r\n\s]+\.([\w\r\n\s]+)\(`
+
+	// Compile the regular expression
+	regex, err := regexp.Compile(pattern)
+	if err != nil {
+		fmt.Println("Error compiling regular expression:", err)
+		os.Exit(1)
+	}
+
+	stableIds := StableIds{
+		StableIds: []string{},
+	}
+	// Perform regexp search on all Go files
+	err = filepath.Walk(folderPath, func(path string, info os.FileInfo, err error) error {
```

Comment:
for readability and maintainability can we give this function a name and not make it inline?